src/main/java/catan/services/util/play/PlayUtil.java

```diff
@@ -0,0 +1,135 @@
+package catan.services.util.play;
+
+import catan.domain.exception.GameException;
+import catan.domain.model.game.GameBean;
+import catan.domain.model.game.GameUserBean;
+import catan.domain.model.game.types.GameStage;
+import catan.domain.model.game.types.GameUserActions;
+import catan.domain.transfer.output.game.AllAvailableActionsDetails;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static catan.services.impl.GameServiceImpl.ERROR_CODE_ERROR;
+
+@Component
+public class PlayUtil {
+    private Logger log = LoggerFactory.getLogger(PlayUtil.class);
+
+    private EndTurnUtil endTurnUtil;
+
+    private static final Gson GSON = new Gson();
+
+    public List<List<String>> getInitialBuildingsSetFromJson(String json) {
+        return GSON.fromJson(json, new TypeToken<List<List<String>>>() {
+        }.getType());
+    }
+
+    public AllAvailableActionsDetails getAllAvailableActions(String json) {
+        return GSON.fromJson(json, AllAvailableActionsDetails.class);
+    }
+
+    public void updateNextMoveOrder(GameBean game) throws GameException {
+        Integer nextMoveNumber;
+        switch (game.getStage()) {
+            case PREPARATION:
+                List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+                nextMoveNumber = endTurnUtil.endTurnImplInPreparationStage(game, initialBuildingsSet);
```

Comment:
it is not clear that in method PlayUtil.updateNextMoveOrder() happens call of endTurnImplInPreparationStage() method.